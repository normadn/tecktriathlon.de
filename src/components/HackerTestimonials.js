import React, { Component } from "react";
import MediaQuery from "react-responsive";

import ContentWrapper from "./ContentWrapper";
import TestimonialList from "./TestimonialList.js";
import TestimonialsBlurb from "./TestimonialsBlurb";
import TestimonialListCompressed from "./TestimonialListCompressed.js";
import TestimonialsBlurbCompressed from "./TestimonialsBlurbCompressed";

import stichler from "../assets/img/testimonials/stichler.jpg";
import hieke from "../assets/img/testimonials/hieke.jpg";

const testimonials = [
  {
    name: "Pascal Stichler",
    project: "Carl",
    projectLink: "https://www.carlfinance.de/",
    image: stichler,
    text:
      "Wir arbeiten bereits seit einem halben Jahr mit Amsel.tech zusammen und planen die Zusammenarbeit weiter fortzusetzen. In den letzten Monaten haben wir einige Web-Apps entwickeln lassen und sind von der zielgerichteten Umsetzung des Teams begeistert."
  },
  {
    name: "Clemens Hieke",
    project: "Konstaplan",
    projectLink: "",
    image: hieke,
    text:
      "Wir sind ein Igenieurbüro und möchten unsere Resourcen nicht mit Software-entwicklung verschwenden. Mit Amsel.tech entwickeln wir aktuell eine Rundumlösung für unseren online Auftritt. Bisher sind wir mit den Leistungen des Teams sehr zufrieden!"
  }
];

class HackerTestimonials extends Component {
  state = {
    currentTestimonial: 0
  };

  nextTestimonial() {
    this.setState({
      currentTestimonial: Math.min(
        this.state.currentTestimonial + 1,
        testimonials.length - 1
      )
    });
  }

  prevTestimonial() {
    this.setState({
      currentTestimonial: Math.max(this.state.currentTestimonial - 1, 0)
    });
  }

  render() {
    const blurb = (
      <span>
        Amsel.tech liefert
        <br />
        Schauen Sie was unsere Kunden zu sagen haben
      </span>
    );
    return (
      <section
        id="referenzen"
        css={{
          overflowX: "hidden"
        }}
      >
        <ContentWrapper>
          <MediaQuery query="screen and (min-width: 1000px)">
            <div
              css={{
                height: "575px",
                position: "relative"
              }}
            >
              <TestimonialsBlurb
                nextTestimonial={() => this.nextTestimonial()}
                prevTestimonial={() => this.prevTestimonial()}
                firstTestimonial={!this.state.currentTestimonial}
                lastTestimonial={
                  this.state.currentTestimonial === testimonials.length - 1
                }
                text={blurb}
              />
              <TestimonialList
                currentTestimonial={this.state.currentTestimonial}
                testimonials={testimonials}
              />
            </div>
          </MediaQuery>
          <MediaQuery query="screen and (max-width: 1000px)">
            <div css={{ margin: "0 0 80px 0" }}>
              <TestimonialsBlurbCompressed text={blurb} />
              <TestimonialListCompressed
                currentTestimonial={this.state.currentTestimonial}
                testimonials={testimonials}
              />
            </div>
          </MediaQuery>
        </ContentWrapper>
      </section>
    );
  }
}

export default HackerTestimonials;
