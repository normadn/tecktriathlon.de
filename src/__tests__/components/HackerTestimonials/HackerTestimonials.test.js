import React from "react";
import HackerTestimonials from "../../../components/HackerTestimonials";
import { shallow } from "enzyme";

describe("HackerTestimonials", () => {
  it("renders correctly", () => {
    expect(shallow(<HackerTestimonials />)).toMatchSnapshot();
  });
});
