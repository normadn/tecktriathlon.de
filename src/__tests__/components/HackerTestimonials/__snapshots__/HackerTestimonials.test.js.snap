// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HackerTestimonials renders correctly 1`] = `
<section
  css={
    Object {
      "overflowX": "hidden",
    }
  }
  id="testimonials"
>
  <_default>
    <MediaQuery
      query="screen and (min-width: 1000px)"
      values={Object {}}
    >
      <div
        css={
          Object {
            "height": "575px",
            "position": "relative",
          }
        }
      >
        <TestimonialsBlurb
          firstTestimonial={true}
          lastTestimonial={false}
          nextTestimonial={[Function]}
          prevTestimonial={[Function]}
          text={
            <span>
              QHacks has something to offer everyone.
              <br />
              See what past hackers thought about the event!
            </span>
          }
        />
        <TestimonialList
          currentTestimonial={0}
          testimonials={
            Array [
              Object {
                "image": "test-file-stub",
                "name": "Amy Lu",
                "project": "Reeltube",
                "projectLink": "https://devpost.com/software/qhacks-v1",
                "text": "In high school, I’d never opened a single computer science course brochure, and now I will be an incoming graduate student in machine learning. QHacks gave me a taster of what tech really was - a creative endeavour with technical tools as the paint brush, and societal-level solution pitching as the frame.",
              },
              Object {
                "image": "test-file-stub",
                "name": "Cooper Leong",
                "project": "ESPECT",
                "projectLink": "https://devpost.com/software/espect",
                "text": "Events such as QHacks have allowed me to delve into topics without fear. I used to find it time consuming and intimidating to learn enough to feel confident about a topic, but now I can tackle it head on. It has been just under a year since my first hackathon, and I can say that it has changed my life for the best, and I look forward to participating in many more!",
              },
              Object {
                "image": "test-file-stub",
                "name": "Benji Christie",
                "project": "",
                "projectLink": "",
                "text": "Our entire team had never deployed a project before so we struggled a lot with getting it up and running. We had to google and learn, try and fail, and repeat, which ended up being the most rewarding experience.",
              },
            ]
          }
        />
      </div>
    </MediaQuery>
    <MediaQuery
      query="screen and (max-width: 1000px)"
      values={Object {}}
    >
      <div
        css={
          Object {
            "margin": "0 0 80px 0",
          }
        }
      >
        <TestimonialsBlurbCompressed
          text={
            <span>
              QHacks has something to offer everyone.
              <br />
              See what past hackers thought about the event!
            </span>
          }
        />
        <TestimonialListCompressed
          currentTestimonial={0}
          testimonials={
            Array [
              Object {
                "image": "test-file-stub",
                "name": "Amy Lu",
                "project": "Reeltube",
                "projectLink": "https://devpost.com/software/qhacks-v1",
                "text": "In high school, I’d never opened a single computer science course brochure, and now I will be an incoming graduate student in machine learning. QHacks gave me a taster of what tech really was - a creative endeavour with technical tools as the paint brush, and societal-level solution pitching as the frame.",
              },
              Object {
                "image": "test-file-stub",
                "name": "Cooper Leong",
                "project": "ESPECT",
                "projectLink": "https://devpost.com/software/espect",
                "text": "Events such as QHacks have allowed me to delve into topics without fear. I used to find it time consuming and intimidating to learn enough to feel confident about a topic, but now I can tackle it head on. It has been just under a year since my first hackathon, and I can say that it has changed my life for the best, and I look forward to participating in many more!",
              },
              Object {
                "image": "test-file-stub",
                "name": "Benji Christie",
                "project": "",
                "projectLink": "",
                "text": "Our entire team had never deployed a project before so we struggled a lot with getting it up and running. We had to google and learn, try and fail, and repeat, which ended up being the most rewarding experience.",
              },
            ]
          }
        />
      </div>
    </MediaQuery>
  </_default>
</section>
`;
